addons/mail/static/src/components/message/message_tests.js

```diff
@@ -983,6 +983,87 @@ QUnit.test('open chat with author on avatar click should be disabled when curren
     );
 });
 
+QUnit.test('Notification Canceled', async function (assert) {
+    assert.expect(8);
+
+    this.data['mail.channel'].records.push({ id: 11 });
+    this.data['res.partner'].records.push({ id: 12, name: "Someone" });
+    this.data['mail.message'].records.push({
+        id: 10,
+        channel_ids: [11],
+        message_type: 'email',
+        model: 'mail.channel',
+        notification_ids: [ 11 ],
+        res_id: 11,
+    });
+    this.data['mail.notification'].records.push({
+        id: 11,
+        failure_type: 'SMTP',
+        mail_message_id: 10,
+        notification_status: 'canceled',
+        notification_type: 'email',
+        res_partner_id: 12,
+    });
+    await this.start();
+    const threadViewer = await this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['insert', {
+            id: 11,
+            model: 'mail.channel',
+        }]],
+    });
+    // Need to wait for messages to be loaded (would be better to do "afterEvent" when implemented)
```

Comment:
It is implemented! (rebase if you don't have it) Just need to add the proper trigger in the business code.

Existing `o-component-message-list-thread-cache-changed` would actually be appropriate if the test was using a message list, but since it's just message here that event won't work unfortunately.

I would actually be fine to have this test as a message list / thread viewer test if it can help reduce the number of such events we introduce in business code.

Reply:
We have to decide, either it has to be splitted in multiple tests like said in your comment above or move it to a message list/thread viewer test ?

Reply:
@alexkuhn opinion about these comments?

Reply:
It's not explained why you need "messages to be loaded". It's a sign that either test makes only sense at a higher-level (thread view) or the reliance on messages to be loaded is the problem to be fixed.

Usually I prefer test on smallest component possible.

Reply:
Then considering the thread viewer has to be passed as prop, I think all of the tests in message that rely on this prop have to be moved in thread viewer (only this one in the context of this PR).

Reply:
And to answer why messages have to be loaded: because otherwise the message does not exist to display it, as simple as that. Data doesn't come from the server magically. (in this cache the thread viewer would load it, but we can't await on it due to no event available in the model)

Reply:
Notification feature does not make sense in a standalone message compnent. It only makes sense in the context of a message displayed inside its origin thread viewer.